misk-action-scopes/src/main/kotlin/misk/scope/ActionScope.kt

```diff
@@ -20,37 +21,58 @@ class ActionScope @Inject internal constructor(
   private val providers: @JvmSuppressWildcards Map<Key<*>, Provider<ActionScopedProvider<*>>>
 ) : AutoCloseable {
   companion object {
-    private val tls = ThreadLocal<LinkedHashMap<Key<*>, Any?>>()
+    private val threadLocalScope = ThreadLocal<LinkedHashMap<Key<*>, Any?>>()
+    private val threadLocalUUID = ThreadLocal<UUID>()
```

Comment:
Rather than having two ThreadLocals, you could create a Key for the UUID and put it in the map?